javascript/sdk/corvy.js

```diff
@@ -1,251 +1,354 @@
+            const response = await this.axiosInstance.post(`${this.API_BASE_URL}/auth`);
+
+            this.bot = response.data.bot;
+        }
+        catch (ex) {
+            console.log(ex)
         }
-      });
 
-      this._log(`Registered command: ${fullPrefix}`);
+        await this._connectSocket();
+        this._setupProcessHandlers();
     }
-  }
 
-  // -------------------------------
-  // Auth
-  // -------------------------------
+    async _connectSocket() {
+        this.socket = new Socket(this.WEBSOCKET_URL, {
+            params: { token: this.token },
+            heartbeatIntervalMs: 25000,
+            transport: WebSocket
+        });
 
-  async login() {
-    try {
-      const res = await this.client.post("/auth");
-      this.user = res.data.bot;
+        this.socket.connect();
 
-      await this.flocks.fetch();
+        this.socket.onOpen(() => {
+            this._log('✅ Socket connection established');
+            this.emit('ready', this.bot);
+        });
+        this.socket.onError((error) => {
+            this._log(`❌ Socket error: ${error.message}`);
+            this.emit('error', new Error(`Socket error: ${error.message}`));
+        });
+        this.socket.onClose((event) => this._log(`Socket connection closed: ${event.code} - ${event.reason}`));
 
-      this.emit("ready", this);
+        this._joinBotChannel();
+    }
 
-      this.startTime = Date.now();
+    _joinBotChannel() {
+        this.channel = this.socket.channel("bot:any", { token: this.token });
 
-      const baseline = await this.client.get("/messages", {
-        params: { cursor: 0 }
-      });
+        this.channel.on("message", (payload) => {
+            if (payload.event === 'new_message') {
+                this._handleNewMessage(payload.message);
+            }
+        });
 
-      this.currentCursor = baseline.data.cursor || 0;
+        this.channel.join()
+            .receive("ok", (response) => {
+                this._log('✅ Successfully joined bot channel');
+            })
+            .receive("error", (response) => {
+                this._log(`❌ Failed to join bot channel: ${JSON.stringify(response)}`);
+                this.emit('error', new Error(`Failed to join bot channel: ${JSON.stringify(response)}`));
+            });
+    }
 
-      this._processMessages();
+    _setupProcessHandlers() {
+        process.on('SIGINT', () => {
+            this._log('\nShutting down...');
+            if (this.socket) {
+                this.socket.disconnect();
+            }
+            process.exit(0);
+        });
+    }
 
-      process.on("SIGINT", () => {
-        console.log("Client shutting down...");
-        process.exit(0);
-      });
-    } catch (err) {
-      this.emit("error", new Error("Error logging in: " + err.message));
-      this._log(`Error logging in: ${err.message}`);
+    checkForStaff(userid, staffs) {
+        return staffs?.includes(userid)
     }
```

Comment:
why is this the type of thing thats built into the sdk? maybe more generic check wrapping but this seems a bit wack

Reply:
For permission checking.. ?

Reply:
for simplicity's sake

Reply:
i mean, if that's what you want, sure